reconstruction/cvt/src/main/java/org/jlab/rec/cvt/measurement/Measurements.java

```diff
@@ -283,11 +297,14 @@ private void addClusters(DetectorType type, List<Surface> clusters) {
 //    }
         
     private List<Surface> getClusterSurfaces(DetectorType type, List<Cluster> clusters, Ray ray) {
-        
-        List<Surface> surfaces = this.getClusterSurfaces(type, clusters, ray);        
```

Comment:
it seems the initial source of the problem is this line, a call to the method from the same method. This was introduced with #351.

Fixing this and reverting to assigning the hemisphere based on the cluster.y instead of with the ray intersection seems to solve the problem.

The ray intersection with the surface doesn't seem to work consistently because when two intersection points are found (for cylindrical surfaces) the first one in the List is not necessarily the closest to the ray origin. This is the case also with intersectionRay